diesel/src/pg/query_builder.rs

```diff
@@ -27,7 +27,10 @@ impl QueryBuilder<Pg> for PgQueryBuilder {
     }
 
     fn push_identifier(&mut self, identifier: &str) -> BuildQueryResult {
-        let escaped_identifier = try!(self.conn.escape_identifier(identifier));
+        let escaped_identifier = match self.conn.escape_identifier(identifier) {
+            Ok(v) => v,
+            Err(e) => return Err(Box::new(e)),
+        };
```

Comment:
This was expanded because the relevant `From` impls that we were relying on are provided for `Box<Error>` and `Box<Error+Send+Sync>`, but not `Box<Error+Send>`. This seemed like the simplest path forward.